### PR TITLE
Introducing Optionals

### PR DESCRIPTION
**Pull request by ingmargoudt**

### CODE REVIEW

File: Mage.Server/src/main/java/mage/server/Session.java [hunk 1]
```diff
@@ -371,12 +371,14 @@ public void fireCallback(final ClientCallback call) {
...
         } catch (HandleCallbackException ex) {
-            User user = UserManager.getInstance().getUser(userId);
-            logger.warn("SESSION CALLBACK EXCEPTION - " + (user != null ? user.getName() : "") + " userId " + userId);
-            logger.warn(" - method: " + call.getMethod());
-            logger.warn(" - cause: " + getBasicCause(ex).toString());
-            logger.trace("Stack trace:", ex);
-            userLostConnection();
+            ex.printStackTrace();
+            UserManager.getInstance().getUser(userId).ifPresent(user-> {
+                logger.warn("SESSION CALLBACK EXCEPTION - " + (user != null ? user.getName() : "") + " userId " + userId);
```
Comment (LevelX2) [inline Mage.Server/src/main/java/mage/server/Session.java]: this is executed only now, if a user is present, so user check != null makes no longer sense.

File: Mage.Server/src/main/java/mage/server/game/GameController.java [hunk 2]
```diff
@@ -332,7 +327,7 @@ public void join(UUID userId) {
         } else {
             joinType = "rejoined";
         }
-        user.addGame(playerId, gameSession);
+        user.get().addGame(playerId, gameSession);
```
Comment (sotovdev) [inline Mage.Server/src/main/java/mage/server/game/GameController.java]: "get" without "isPresent"?

Comment (ingmargoudt) [inline Mage.Server/src/main/java/mage/server/game/GameController.java]: Correct, thats incorrect :)

Comment (sotovdev) [inline Mage.Server/src/main/java/mage/server/game/GameController.java]: You fixed this issue in "small fixes after comments" commit or somewhere else?

**Comment by ingmargoudt** [comment]
Yes i corrected that in the latest commit

Verstuurd vanaf mijn iPhone

**Pull request merged**